### PR TITLE
Implement database and express configuration

### PR DESCRIPTION
**Pull request by maaaashin324**
Implemented database and express configuration. But there is still a problem I want to fix for routing. In order to route post method, I need to write 'router.post('/:id', async (req, res) => {', but this is weird.

I wanna move forward somehow, so I made this PR, but I'll fix this problem. If you know why I'll appreciate it if you let me know it.

And I changed the directory structure according to express documentation.

### CODE REVIEW

**Comment by maaaashin324** [comment]
I focused on message table, so after figuring out what information is necessary for flight plans, I'll modify some files for flight plans table.

File: app.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+app.use(logger('dev'));
+app.use(bodyParser.json());
+// you cannot send nested objects to a server.
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(express.static(path.join(__dirname, '../public')));
+
+// router with db setup
+const config = require('./config');
+const middlewares = require('./middlewares/index')(config);
+const messages = require('./routes/messages')(middlewares);
```
Comment (FelixKirmse) [inline app.js]: Try to have all requires on top of the file.

File: package.json [hunk 2]
```diff
@@ -4,13 +4,18 @@
     "description": "This repository is made by students of Code Chrysalis",
     "main": "index.js",
     "scripts": {
-        "test": "echo \"Error: no test specified\" && exit 1"
+        "test": "echo \"Error: no test specified\" && exit 1",
+        "migrate": "knex migrate:latest --knexfile ./middlewares/db/knexfile.js",
+        "rollback": "knex migrate:rollback --knexfile ./middlewares/db/knexfile.js",
+        "start": "node index.js"
```
Comment (FelixKirmse) [inline package.json]: Highly suggest a postinstall that runs the migrate script!

File: routes/messages/index.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    try {
+      const messages = await db.messages.listMessage(id);
+      res.status(200).json(messages);
+    } catch (error) {
+      res.send(500, error.message);
+    }
+  });
+
+  router.post('/:id', async (req, res) => {
```
Comment (FelixKirmse) [inline routes/messages/index.js]: I am not sure what your problem you were describing it here? Do you have to use /:id when testing? I checked out the project and had it say just `router.post('/', async (req, res) => {` and it worked.

But I noticed that your error handling is not working correctly! If I post nonsense data, I get back a 201 status and the "insertedMessage" will be an error from knex.

**Review by FelixKirmse — CHANGES_REQUESTED** [review]
You are missing a README and steps on what is needed to run your project! For example: "Needs to create database by running `createdb rescue_drone` and having to run yarn migrate manually.

Also check [this](https://stackoverflow.com/a/27344933) out if you don't want to bother users with creating a database. 

Delete `server/index.js` and `server/services/db/knexfile.js` if you don't need them.

File: routes/messages/index.js (hunk 3 above)
Comment (maaaashin324) [reply routes/messages/index.js]: Yes, it is just a problem I'm struggled with. If there is '/:id', I can get the data and insert data into the database. If I delete it, I cannot get the data.

File: routes/messages/index.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+const express = require('express');
+
+const router = express.Router();
+
+module.exports = (middlewares) => {
+  const { db } = middlewares;
+
+  router.get('/:id', async (req, res) => {
+    const id = req.query;
```
Comment (FelixKirmse) [inline routes/messages/index.js]: I suggest changing this to `const { id } = req.params` as `req.params` should be used when you are trying to resolve route parameters.

File: routes/messages/index.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    try {
+      const insertedMessage = await db.messages.createMessage(message);
+      res.status(201).json(insertedMessage);
+    } catch (error) {
+      res.send(500, error.message);
+    }
+  });
+
+  router.put('/:id', async (req, res) => {
+    const id = req.query;
```
Comment (FelixKirmse) [inline routes/messages/index.js]: I suggest changing this to `const { id } = req.params` as `req.params` should be used when you are trying to resolve route parameters.

File: routes/messages/index.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    try {
+      const updatedMessage = await db.messages.updateMessage({ id, updateContents });
+      res.status(202).json(updatedMessage);
+    } catch (error) {
+      res.send(500, error.message);
+    }
+  });
+
+  router.delete('/:id', async (req, res) => {
+    const { id } = req.query;
```
Comment (FelixKirmse) [inline routes/messages/index.js]: I suggest changing this to `const { id } = req.params` as `req.params` should be used when you are trying to resolve route parameters.

**Review by FelixKirmse — CHANGES_REQUESTED** [review]
Found some more =)

File: routes/messages/index.js (hunk 6 above)
Comment (FelixKirmse) [reply routes/messages/index.js]: Also I realized you are only destructuring here and not in the others, might be the cause of some of your issues?

**Review by alpenfritz — APPROVED** [review]

**Comment by maaaashin324** [comment]
After connecting user front end part and backend, we'll make a PR

**Pull request closed without merging**